src/components/command-bar/commands/search/unified-search/components/dialog-body/helpers/get-algolia-product-filter-string.ts

```diff
@@ -0,0 +1,32 @@
+import { ProductSlug } from 'types/products'
+
+/**
+ * Given an optional product slug,
+ *
+ * Return an Algolia `filter` string that will filter for search objects
+ * of any type (`docs`, `tutorial`, or `integration`) that match the
+ * specified product slug.
+ *
+ * Note: intended for use with our unified search indices, which are
+ * named `<env>_DEVDOT_omni` in Algolia.
+ */
+export function getAlgoliaProductFilterString(
+	productSlug?: ProductSlug
+): string {
+	let filterString = ''
+
+	if (productSlug) {
+		/**
+		 * The edition:hcp only applies to `tutorials` records, which will
+		 * never have products:hcp, but we can't apply complex filters
+		 * via the Algolia filters API parameter, so we keep it simple here.
+		 * Ref: https://www.algolia.com/doc/api-reference/api-parameters/filters/
+		 *
+		 * TODO: look into how tutorials library handles this? Does it use
+		 * the filters parameter?
+		 */
+		filterString = `products:${productSlug} OR edition:${productSlug}`
```

Comment:
Looks like we do a [similar thing here](https://github.com/hashicorp/dev-portal/blob/5d49306ec4a582efa590f1fee0f81030cfb08c01/src/components/command-bar/commands/search/components/tutorials-tab-contents/index.tsx#L20)

Should we be specific about `edition:hcp` here? Since none of the other products use that?

Reply:
Ah great call! Will update this to only do `OR edition:${productSlug}` if `productSlug === 'hcp'` 👍